br/pkg/lightning/restore/restore.go

```diff
@@ -2407,6 +2461,56 @@ func saveCheckpoint(rc *Controller, t *TableRestore, engineID int32, chunk *chec
 	}
 }
 
+func filterColumns(columnNames []string, extendData mydump.ExtendColumnData, ignoreColsMap map[string]struct{}, tableInfo *model.TableInfo) ([]string, []string, []types.Datum) {
+	extendCols, extendVals := extendData.Columns, extendData.Values
+	extendColsMap := make(map[string]struct{})
+	for _, extendCol := range extendCols {
+		extendColsMap[extendCol] = struct{}{}
+	}
+	for c := range ignoreColsMap {
```

Comment:
i think `ignore column` should not mix with `extend column`
- we can only ingore exist column
- we can only extend column that didn't exist before